hibernate-core/src/main/java/org/hibernate/metamodel/internal/AttributeFactory.java

```diff
@@ -220,8 +220,7 @@ public Property getPropertyMapping() {
 				final Component component = (Component) typeContext.getHibernateValue();
 				final EmbeddableTypeImpl<Y> embeddableType;
 
-				if ( component.getComponentClass() != null
-						|| component.getComponentClassName() != null ) {
+				if ( component.getComponentClassName() != null ) {
```

Comment:
One thing I noticed is that calling `Component#getComponentClassName()` will fail for map-mode mappings because no component-class-name has been set.  I could have reversed checks and used `&&` instead, I guess its a matter more of preference as the end-result is the same.